common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsManager.kt

```diff
@@ -0,0 +1,101 @@
+package com.revenuecat.purchases.common.diagnostics
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.Dispatcher
+import com.revenuecat.purchases.common.verboseLog
+import java.io.IOException
+
+class DiagnosticsManager(
+    private val diagnosticsFileHelper: DiagnosticsFileHelper,
+    private val diagnosticsAnonymizer: DiagnosticsAnonymizer,
+    private val backend: Backend,
+    private val diagnosticsDispatcher: Dispatcher,
+    private val sharedPreferences: SharedPreferences
+) {
+    companion object {
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        const val CONSECUTIVE_FAILURES_COUNT_KEY = "consecutive_failures_count"
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        const val MAX_NUMBER_RETRIES = 3
+
+        fun initializeSharedPreferences(context: Context): SharedPreferences =
+            context.getSharedPreferences(
+                "com_revenuecat_purchases_${context.packageName}_preferences_diagnostics",
+                Context.MODE_PRIVATE
+            )
+    }
+
+    fun syncDiagnosticsFileIfNeeded() {
+        enqueue {
+            try {
+                val diagnosticsList = diagnosticsFileHelper.readDiagnosticsFile()
+                val diagnosticsCount = diagnosticsList.size
+                if (diagnosticsCount == 0) {
+                    verboseLog("No diagnostics to sync.")
+                    return@enqueue
+                }
+                backend.postDiagnostics(
+                    diagnosticsList = diagnosticsList,
+                    onSuccessHandler = {
+                        verboseLog("Synced diagnostics file successfully.")
+                        clearConsecutiveNumberOfErrors()
+                        diagnosticsFileHelper.cleanSentDiagnostics(diagnosticsCount)
+                    },
+                    onErrorHandler = { error, shouldRetry ->
+                        verboseLog("Error syncing diagnostics file: $error. Should retry: $shouldRetry")
+                        if (shouldRetry) {
+                            val currentRetryCount = increaseConsecutiveNumberOfErrors()
+                            if (currentRetryCount >= MAX_NUMBER_RETRIES) {
+                                resetDiagnosticsStatus()
+                            }
+                        } else {
+                            resetDiagnosticsStatus()
+                        }
+                    }
+                )
+            } catch (e: IOException) {
+                verboseLog("Error syncing diagnostics file: $e")
```

Comment:
We need diagnostics for diagnostics...
In all seriousness, I'm curious why all these are verbose?

Reply:
+1

Reply:
I guess it's to prevent noise for folks actually debugging, since this isn't that much of a developer-facing feature (other than opting in)

Reply:
like, if we had `error` level for these, folks might get concerned but it won't affect their apps in any way

Reply:
Yeah, that's my thought on this. Since this is a non-critical feature, I thought we don't want to polute the logs with it. As mentioned before, lmk if you prefer to send errors/warnings and I can change it, not opposed to that approach either.

Reply:
Yeah that makes sense, wanted to make sure I understood.